native/src/main/java/io/ballerina/lib/data/csvdata/FromString.java

```diff
@@ -0,0 +1,380 @@
+                case TypeTags.BOOLEAN_TAG -> stringToBoolean(value);
+                case TypeTags.NULL_TAG -> stringToNull(value, config);
+                case TypeTags.FINITE_TYPE_TAG -> stringToFiniteType(value, (FiniteType) expType, config);
+                case TypeTags.UNION_TAG -> stringToUnion(string, (UnionType) expType, config, false);
+                case TypeTags.JSON_TAG, TypeTags.ANYDATA_TAG ->
+                        stringToUnion(string, JSON_TYPE_WITH_BASIC_TYPES, config, true);
+                case TypeTags.TYPE_REFERENCED_TYPE_TAG ->
+                        fromStringWithType(string, ((ReferenceType) expType).getReferredType(), config);
+                case TypeTags.INTERSECTION_TAG ->
+                        fromStringWithType(string, ((IntersectionType) expType).getEffectiveType(), config);
+                default -> returnError(value, expType.toString());
+            };
+        } catch (NumberFormatException | ParseException e) {
+            return returnError(value, expType.toString());
+        }
+    }
+
+    private static Object stringToFiniteType(String value, FiniteType finiteType, CsvConfig config) {
+        return finiteType.getValueSpace().stream()
+                .filter(finiteValue -> !(convertToSingletonValue(value, finiteValue, config) instanceof BError))
+                .findFirst()
+                .orElseGet(() -> returnError(value, finiteType.toString()));
+    }
+
+    private static Object convertToSingletonValue(String str, Object singletonValue, CsvConfig config) {
+        String singletonStr = String.valueOf(singletonValue);
+        if (str.equals(singletonStr)) {
+            return fromStringWithType(StringUtils.fromString(str), TypeUtils.getType(singletonValue), config);
+        } else {
+            return returnError(str, singletonStr);
+        }
+    }
+
+    private static Long stringToInt(String value, CsvConfig config) throws NumberFormatException, ParseException {
+        Number number = parseNumberValue(value, config);
+        if (isIntegerValue(value, number, config.locale)) {
+            return number.longValue();
+        }
+        throw new NumberFormatException();
+    }
+
+    private static int stringToByte(String value, CsvConfig config) throws NumberFormatException, ParseException {
+        Number number = parseNumberValue(value, config);
+        if (isIntegerValue(value, number, config.locale)) {
+            int intValue = parseNumberValue(value, config).intValue();
+            if (isByteLiteral(intValue)) {
+                return intValue;
+            }
+        }
+        throw new NumberFormatException();
+    }
+
+    private static long stringToSigned8Int(String value, CsvConfig config) throws NumberFormatException,
+            ParseException {
+        Number number = parseNumberValue(value, config);
+        if (isIntegerValue(value, number, config.locale)) {
+            long intValue = parseNumberValue(value, config).longValue();
```

Comment:
Can't we directly use number here
```suggestion
            long intValue = number.longValue();
```

Reply:
Check other places as well